engine.py

```diff
@@ -515,52 +510,112 @@ def has_ui(self):
     ##########################################################################################
     # logging
 
-    def log_debug(self, msg):
+    def _emit_log_message_FUTURE(self, handler, record):
         """
-        Log debug to the Maya script editor
+        Called by the engine to log messages in Maya script editor.
+        All log messages from the toolkit logging namespace will be passed to this method.
+
+        .. note:: This method will be called by the new logging system introduced in core v0.18.
+                  It will be enabled in January 2017 by removing the method name suffix "_FUTURE".
 
-        :param msg: The message to log
+        :param handler: Log handler that this message was dispatched from.
+        :type handler: :class:`~python.logging.LogHandler`
+        :param record: Standard python logging record.
+        :type record: :class:`~python.logging.LogRecord`
         """
-        global g_last_message_time
 
-        # get the time stamps
-        prev_time = g_last_message_time
-        current_time = time.time()
+        # Do not log debug messages when debug logging setting is off.
+        if record.levelno < logging.INFO and not self.get_setting("debug_logging", False):
+            return
+
+        # Give a standard format to the message.
+        msg = "Shotgun %s" % handler.format(record)
+
+        # Select Maya display function to use according to the logging record level.
+        if record.levelno < logging.WARNING:
+            fct = OpenMaya.MGlobal.displayInfo
+        elif record.levelno < logging.ERROR:
+            fct = OpenMaya.MGlobal.displayWarning
+        else:
+            fct = OpenMaya.MGlobal.displayError
 
-        # update global counter
-        g_last_message_time = current_time
+        # Display the message in Maya script editor in a thread safe manner.
+        self.async_execute_in_main_thread(fct, msg)
 
+    def log_debug(self, msg):
+        """
+        Logs debug messages in Maya script editor.
+
+        .. note:: This method is called by the legacy logging system.
+                  It will be removecd in January 2017.
+
+        :param msg: Debug message to log.
+        """
+
+        # Do not log debug messages when debug logging setting is off.
         if not self.get_setting("debug_logging", False):
             return
 
-        msg = "Shotgun Debug [%0.3fs]: %s" % ((current_time-prev_time), msg)
+        # Give a standard format to the message.
+        # The message parameter uses legacy format "%(basename)s: %(message)s".
+        # Make it look like core v0.18 format "[%(levelname)s %(basename)s] %(message)s"
+        msg = "Shotgun DEBUG %s" % msg
```

Comment:
Like @josh-t commented in another CR, if timing needs to be checked, it is already logged better in `tk-maya.log`. No need to clog the tk-maya logging methods with it.

Though, to be fair and cover all the cases, tk-maya engine should be updated to use a logger rather than calling directly the `log_...` methods.